gdk4-win32/src/win32_display.rs

```diff
@@ -24,4 +26,64 @@ impl Win32Display {
             }
         }
     }
+
+    #[doc(alias = "gdk_win32_display_add_filter")]
+    pub fn add_filter<F>(&self, filter_func: F) -> Win32DisplayFilterHandle<F>
+    where
+        F: Fn(&Self, &mut MSG, &mut i32) -> Win32MessageFilterReturn + 'static,
+    {
+        let f = Box::into_raw(Box::new(filter_func)) as *mut _;
+        unsafe {
+            ffi::gdk_win32_display_add_filter(self.to_glib_none().0, Some(trampoline::<F>), f);
+        }
+
+        Win32DisplayFilterHandle {
+            display: self.clone(),
```

Comment:
If we use this approach for handler/drop, I would rather keep a weak reference to self. How to do that?